src/map/battlefield.cpp

```diff
@@ -463,9 +463,10 @@ bool CBattlefield::RemoveEntity(CBaseEntity* PEntity, uint8 leavecode)
                 if (static_cast<CPetEntity*>(PEntity)->isAlive() && PEntity->PAI->IsSpawned())
                     static_cast<CPetEntity*>(PEntity)->Die();
 
-                m_AllyList.erase(std::remove_if(m_AllyList.begin(), m_AllyList.end(), check), m_AllyList.end());
-                GetZone()->DeletePET(PEntity);
-                delete PEntity;
+                if (m_AllyList.size() > 0)
```

Comment:
You should not need to check the size of the object as erase with begin == end will result in no-op - https://en.cppreference.com/w/cpp/container/vector/erase

Reply:
I did leave it off but it was crashing on it without the if statement..  i can try it again